core/iwasm/interpreter/wasm_loader.c

```diff
@@ -11180,11 +11181,40 @@ wasm_loader_prepare_bytecode(WASMModule *module, WASMFunction *func,
                 /* Check stack values match return types by comparing tag param
                  * types with stack cells */
                 uint8 *frame_ref = loader_ctx->frame_ref;
+#if WASM_ENABLE_GC != 0
+                WASMRefTypeMap *frame_reftype_map =
+                    loader_ctx->frame_reftype_map;
+                uint32 frame_reftype_map_num = loader_ctx->reftype_map_num;
+                param_reftype_maps = tag_type->ref_type_maps;
+                /* For tag_type function, it shouldn't have result_count = 0 */
+                param_reftype_map_count = tag_type->ref_type_map_count;
+                param_count = (int32)tag_type->param_count;
+#endif
+
                 for (int tti = (int32)tag_type->param_count - 1; tti >= 0;
                      tti--) {
+#if WASM_ENABLE_GC != 0
+                    local_type = tag_type->types[tti];
+                    local_idx = tti;
+                    bool is_type_multi_byte =
+                        wasm_is_type_multi_byte_type(local_type);
+                    WASMRefType *ref_type = NULL;
+                    if (is_type_multi_byte) {
```

Comment:
Seems no need to check again? GET_LOCAL_REFTYPE macro already checks `if (wasm_is_type_multi_byte_type(local_type))`